src/test/test-condition.c

```diff
@@ -125,6 +127,68 @@ static void test_condition_test_path(void) {
         condition_free(condition);
 }
 
+static void test_condition_test_control_group_controller(void) {
+        Condition *condition;
+        CGroupMask system_mask;
+        CGroupController controller;
+        char *controller_name;
+
+        // Invalid controllers are ignored
+        condition = condition_new(CONDITION_CONTROL_GROUP_CONTROLLER, "thisisnotarealcontroller", false, false);
+        assert_se(condition);
+        assert_se(condition_test(condition));
+        condition_free(condition);
+
+        condition = condition_new(CONDITION_CONTROL_GROUP_CONTROLLER, "thisisnotarealcontroller", false, true);
+        assert_se(condition);
+        assert_se(!condition_test(condition));
+        condition_free(condition);
+
+        assert_se(cg_mask_supported(&system_mask) >= 0);
+
+        // Individual valid controllers one by one
+        for (controller = 0; controller < _CGROUP_CONTROLLER_MAX; controller++) {
+                controller_name = (char*)cgroup_controller_to_string(controller);
```

Comment:
it's mean to cast the "const" away... please assign this to a "const char*" variable here, please.